File: src/types/tv.ts
export interface TvSeries {
  id: string;
  tmdb_id?: number | null;
  title: string;
  original_title?: string | null;
  year?: number | null;
  description?: string | null;
  poster_path?: string | null;
  backdrop_path?: string | null;
  genres: string[];
  rating?: number | null;
  created_at: string;
  updated_at: string;
}

export interface TvSeason {
  id: string;
  series_id: string;
  season_number: number;
  name: string;
  overview?: string | null;
  poster_path?: string | null;
  episode_count: number;
  created_at: string;
}

export interface TvEpisode {
  id: string;
  series_id: string;
  season_id: string;
  season_number: number;
  episode_number: number;
  title: string;
  overview?: string | null;
  still_path?: string | null;
  air_date?: string | null;
  duration_seconds?: number | null;
  rating?: number | null;
  created_at: string;
  updated_at: string;
}

export interface EpisodeWithMedia {
  episode: TvEpisode;
  media_path?: string | null;
  media_id?: string | null;
  progress_seconds: number;
  duration_seconds: number;
  completed: boolean;
}

export interface SeasonWithEpisodes {
  season: TvSeason;
  episodes: EpisodeWithMedia[];
}

export interface SeriesDetails {
  series: TvSeries;
  seasons: SeasonWithEpisodes[];
  total_episodes: number;
}

File: src-tauri/src/domain/tv.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use super::MetadataStatus;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TvSeries {
    pub id: Uuid,
    pub tmdb_id: Option<i64>,
    pub title: String,
    pub original_title: Option<String>,
    pub year: Option<u16>,
    pub description: Option<String>,
    pub poster_path: Option<String>,
    pub backdrop_path: Option<String>,
    pub genres: Vec<String>,
    pub rating: Option<f32>,
    pub metadata_provider_id: Option<String>,
    pub metadata_status: MetadataStatus,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TvSeason {
    pub id: Uuid,
    pub series_id: Uuid,
    pub season_number: u32,
    pub name: String,
    pub overview: Option<String>,
    pub poster_path: Option<String>,
    pub episode_count: u32,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TvEpisode {
    pub id: Uuid,
    pub series_id: Uuid,
    pub season_id: Uuid,
    pub season_number: u32,
    pub episode_number: u32,
    pub title: String,
    pub overview: Option<String>,
    pub still_path: Option<String>,
    pub air_date: Option<String>,
    pub duration_seconds: Option<u32>,
    pub rating: Option<f32>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EpisodeWithMedia {
    pub episode: TvEpisode,
    pub media_path: Option<String>,
    pub media_id: Option<Uuid>,
    pub progress_seconds: u32,
    pub duration_seconds: u32,
    pub completed: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SeasonWithEpisodes {
    pub season: TvSeason,
    pub episodes: Vec<EpisodeWithMedia>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SeriesDetails {
    pub series: TvSeries,
    pub seasons: Vec<SeasonWithEpisodes>,
    pub total_episodes: u32,
}
